### DISCUSSION TITLE
built-in WebView

### DISCUSSION THREAD

**Discussion by Marco-9456** [JunkFood02/Seal#433] [General]
Hello @JunkFood02 

Since you've added a built-in WebView in Seal, I tested it on some websites like: 
- Youtube
- Instagram
- Facebook
- Youtube-Music
- SoundCloud
- Crunchyroll

It worked great on all of them but not on Crunchyroll and Instagram, on Crunchyroll the site just keeps loading, and interacting with it is not working as you can see in the following Screen record:

https://user-images.githubusercontent.com/41828058/208295913-e551faac-378a-4122-b2cc-98fc74d82dd3.mp4

For Instagram the site is just grayed out:

https://user-images.githubusercontent.com/41828058/208295977-e09785b2-7ad9-4f74-ad7e-ee031f175322.mp4

I thought the user agent you were using is causing the problem so I changed it but it didn't solve anything.

**Comment by JunkFood02** [comment]
I think the problem should be about the webview and there's maybe not a good solution for this. Because we're only able to obtain the (limited) cookies from the default webview but not the powerful chrome one.

Instagram works good on my side. For the problem of crunchyroll, a possible work-around is to manually input the URL of log-in page (sth like `https://sso.crunchyroll.com/***auth***`)

**Reply by BobbyESP** [reply]
With chrome custom tabs cookies can be retrieved?

**Reply by JunkFood02** [reply]
I don't think so. `CookieManager` is only work for WebView, and letting an app to obtain cookies from user's browser is way too dangerous

**Reply by BobbyESP** [reply]
https://stackoverflow.com/questions/36728869/chrome-custom-tabs-cookiemanager

Not possible with Chrome custom tabs.

**Reply by Marco-9456** [reply]
Thanks, the workaround you've mentioned worked,but I've noticed a bug when pasting a long URL the `generat new cookies` option disappear partly as you can see in this Screenshot:
![Screenshot_20221219_012320_Seal](https://user-images.githubusercontent.com/41828058/208327349-f790b4bf-6fa5-4b67-a7ff-95587aa17f84.jpg)

Edit: After the latest changes you've made the bug i mentioned in this reply is no longer there 👍, and Instagram is now working
